### PR TITLE
feat: support beta versions with m1sta's beta API

### PR DESCRIPTION
**Pull request by beer-psi**

### CODE REVIEW

File: sepfinder.py [hunk 1]
```diff
@@ -93,11 +93,27 @@ def on_text(update, ctx):
...
+            ctx.bot_data['ipswme_failed'] = False
+        else:
+            device = {
+                'name': device['name'],
+                'identifier': device['identifier'],
+                'boards': device['boards'],
+                'firmwares': [],
+            }
+            update.message.reply_text('Unable to communicate with ipsw.me API.')
+            ctx.bot_data['ipswme_failed'] = True
```
Comment (ghost) [inline sepfinder.py]:
```suggestion
            ctx.user_data['ipswme_failed'] = True
```
It shouldn't use `bot_data`, could cause issues if multiple users are using the bot at once.

File: sepfinder.py [hunk 2]
```diff
@@ -93,11 +93,27 @@ def on_text(update, ctx):
...
+
+        if rb.ok:
+            device_beta = rb.json()
+            device['firmwares'] += device_beta
+        else:
+            update.message.reply_text('Unable to communicate with the beta API.')
+            if ctx.bot_data['ipswme_failed']:
```
Comment (ghost) [inline sepfinder.py]:
```suggestion
            if ctx.user_data['ipswme_failed']:
```

File: sepfinder.py [hunk 3]
```diff
@@ -249,7 +265,7 @@ def show_firmware_menu(update, ctx):
             'Invalid state. Please start over using /start.', reply_markup=ReplyKeyboardRemove(),
         )
 
-    firmwares = [x for x in ctx.user_data['device']['firmwares'] if x['signed']]
+    firmwares = [x for x in ctx.user_data['device']['firmwares'] if 'signed' in x and x['signed']]
```
Comment (ghost) [inline sepfinder.py]:
```suggestion
    firmwares = [x for x in ctx.user_data['device']['firmwares'] if x.get('signed')]
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: sepfinder.py (hunk 1 above)
Comment (ghost) [reply sepfinder.py]: Hmm, technically I use it for the devices dict but I probably shouldn't be doing that either, at least not as-is. The cached response is not currently reused for new requests and it just runs the risk of a race condition where it overwrites one user's successful response with a failed response another user got in meantime.

Either way, even if I do properly cache the device list in `bot_data` in the future, it should probably always retry if there's no successful cached response so it makes sense for this to stay `user_data`.

**Comment by beer-psi** [comment]
All suggested changes have been made.

**Review by ghost — APPROVED** [review]

**Pull request merged**